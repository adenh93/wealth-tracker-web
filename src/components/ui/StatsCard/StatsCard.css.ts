import { styled } from '@mui/material/styles'
import { Box } from '@mui/material'

export const Wrapper = styled(Box)({
  display: 'flex',
  alignItems: 'center',
  columnGap: '1.8rem',
})

export const ContentWrapper = styled(Box)({
  display: 'flex',
  flexDirection: 'column',
  rowGap: '0.2rem',
})
