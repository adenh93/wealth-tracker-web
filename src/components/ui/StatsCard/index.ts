export { default } from './StatsCard'
export type { StatsCardProps } from './StatsCard'