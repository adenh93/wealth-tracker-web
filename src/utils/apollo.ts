import { ApolloError } from '@apollo/client'

export const extractInputErrorField = <FormType>(
  error: ApolloError
): keyof FormType | null => {
  if (error.graphQLErrors.length)
    return error.graphQLErrors[0].extensions?.field as keyof FormType
  return null
}
