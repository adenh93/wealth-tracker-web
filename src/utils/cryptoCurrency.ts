import { CMC_LOGO_URL } from '../constants'

export const getCMCLogoUrl = (id: number): string => `${CMC_LOGO_URL}/${id}.png`
