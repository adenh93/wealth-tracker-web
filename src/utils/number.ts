const defaultOpts: Intl.NumberFormatOptions = {
  maximumFractionDigits: 2,
  notation: 'standard',
  signDisplay: 'never',
}

export const formatNumber = (
  value: number,
  opts: Intl.NumberFormatOptions = defaultOpts
): string =>
  new Intl.NumberFormat('en-US', {
    ...opts,
  }).format(value)

export const formatCurrency = (
  value: number,
  opts: Intl.NumberFormatOptions = defaultOpts
): string =>
  formatNumber(value, { ...opts, style: 'currency', currency: 'AUD' })

export const formatPercent = (
  value: number,
  opts: Intl.NumberFormatOptions = defaultOpts
): string => formatNumber(value, { ...opts, style: 'percent' })
