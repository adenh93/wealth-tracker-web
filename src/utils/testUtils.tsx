import { ReactElement } from 'react'
import { ThemeProvider } from '@mui/material'
import { render as tlRender, RenderResult } from '@testing-library/react'
import theme from '../theme'

export const render = (component: ReactElement): RenderResult =>
  tlRender(<ThemeProvider theme={theme}>{component}</ThemeProvider>)
