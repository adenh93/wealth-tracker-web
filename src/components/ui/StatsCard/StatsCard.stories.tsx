import { Story } from '@storybook/react'
import { green, lightGreen } from '@mui/material/colors'
import AttachMoneyIcon from '@mui/icons-material/AttachMoney'
import StatsCard, { StatsCardProps } from './StatsCard'
import { Typography } from '@mui/material'
import ProfitLossLabel from '../../ProfitLossLabel'

const config = {
  title: 'Components/UI/StatsCard',
  component: StatsCard,
}

const Template: Story<StatsCardProps> = (args) => (
  <StatsCard {...args}>
    <ProfitLossLabel value={210.56} type="currency" variant="subtitle1" />
    <Typography variant="body1">A$87,000.56</Typography>
    <Typography variant="body2">Total Net Worth</Typography>
  </StatsCard>
)

export const Primary = Template.bind({})

Primary.args = {
  Icon: AttachMoneyIcon,
  iconColor: green[500],
  iconBackgroundColor: lightGreen[100],
}

export default config
