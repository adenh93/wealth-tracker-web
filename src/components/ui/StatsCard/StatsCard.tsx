import { Avatar, Paper } from '@mui/material'
import { FC } from 'react'
import { Wrapper, ContentWrapper } from './StatsCard.css'

export interface StatsCardProps {
  Icon: any
  iconColor: string
  iconBackgroundColor: string
}

const StatsCard: FC<StatsCardProps> = ({
  Icon,
  iconColor,
  iconBackgroundColor,
  children,
}) => (
  <Paper sx={{ padding: 4.5, borderRadius: '0.5rem' }}>
    <Wrapper>
      <Avatar sx={{ bgcolor: iconBackgroundColor, width: 85, height: 85 }}>
        <Icon sx={{ fill: iconColor, fontSize: '60px' }} />
      </Avatar>
      <ContentWrapper>{children}</ContentWrapper>
    </Wrapper>
  </Paper>
)

export default StatsCard
