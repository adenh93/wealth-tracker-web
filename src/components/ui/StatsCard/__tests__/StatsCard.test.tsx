import AttachMoneyIcon from '@mui/icons-material/AttachMoney'
import { green, lightGreen } from '@mui/material/colors'
import { render } from '../../../../utils/testUtils'
import StatsCard from '../StatsCard'

describe('StatsCard component tests', () => {
  test('matches snapshot', () => {
    const { asFragment } = render(
      <StatsCard
        Icon={AttachMoneyIcon}
        iconColor={green[500]}
        iconBackgroundColor={lightGreen[100]}
      >
        Test
      </StatsCard>
    )

    expect(asFragment).toMatchSnapshot()
  })
})
