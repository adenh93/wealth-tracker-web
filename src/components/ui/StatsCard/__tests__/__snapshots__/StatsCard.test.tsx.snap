// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatsCard component tests matches snapshot 1`] = `[Function]`;
